src/components/views/rooms/Autocomplete.js

```diff
@@ -163,26 +163,17 @@ export default class Autocomplete extends React.Component {
     }
 
     // called from MessageComposerInput
-    onUpArrow(): ?Completion {
+    moveSelection(delta): ?Completion {
         const completionCount = this.countCompletions();
-        // completionCount + 1, since 0 means composer is selected
-        const selectionOffset = (completionCount + 1 + this.state.selectionOffset - 1)
-            % (completionCount + 1);
-        if (!completionCount) {
-            return null;
+        if (completionCount === 0) {
+            // completionCount 0 means autocomplete inactive
+            return null
         }
-        this.setSelection(selectionOffset);
-    }
 
-    // called from MessageComposerInput
-    onDownArrow(): ?Completion {
-        const completionCount = this.countCompletions();
-        // completionCount + 1, since 0 means composer is selected
-        const selectionOffset = (this.state.selectionOffset + 1) % (completionCount + 1);
-        if (!completionCount) {
-            return null;
-        }
-        this.setSelection(selectionOffset);
+        // selectionOffset 0 means autocomplete inactive, while 1 means 0th item selected, hence the -1/+1
+        let index = this.state.selectionOffset - 1;
+        index = (index + delta + completionCount) % completionCount;
+        this.setSelection(index + 1);
```

Comment:
Won't this prevent from ever going back selectionOffset=0 again? E.g. you'll just be able to cycle through pills without having the step with just the text you typed?

Reply:
You're right, good catch, I see the point behind the `%(completionCount+1)` now!

I've updated the above to:
```javascript
// Note: selectionOffset 0 represents the unsubstituted text, while 1 means first pill selected
const index = (this.state.selectionOffset + delta + completionCount + 1) % (completionCount + 1);
this.setSelection(index);
```